source/agora/test/Base.d

```diff
@@ -974,108 +972,61 @@ public APIManager makeTestNetwork (APIManager : TestAPIManager = TestAPIManager)
         ban_duration: 300
     };
 
-    Config makeConfig (NodeConfig self, NodeConfig[] node_confs)
+    Config makeConfig (size_t idx, NodeConfig self, NodeConfig[] node_confs)
     {
         auto other_nodes =
             node_confs
                 .filter!(conf => conf != self)
                 .map!(conf => conf.address);
 
-        Config conf =
-        {
-            banman : ban_conf,
-            node : self,
-            network : test_conf.configure_network ? assumeUnique(other_nodes.array) : null
-        };
-
-        return conf;
-    }
-
-    // for discovery testing: only 1 node in the 'network' section, rest is discovered
-    Config makeMinimalNetwork (size_t idx, NodeConfig self, NodeConfig[] node_confs)
-    {
         auto prev_node = idx == 0 ? node_confs[$ - 1] : node_confs[idx - 1];
 
-        Config conf =
+        string[] network;
+        if (test_conf.configure_network)
         {
-            banman : ban_conf,
-            node : self,
-            network : test_conf.configure_network ? [prev_node.address] : null
-        };
-
-        return conf;
-    }
-
-    // for discovery testing: full quorum, but only 1 node in the 'network' section
-    Config makeMinimalNetQuorum (size_t idx, NodeConfig self, NodeConfig[] node_confs)
-    {
-        auto prev_node = idx == 0 ? node_confs[$ - 1] : node_confs[idx - 1];
-
-        node_confs.each!(conf => assert(conf.is_validator));
+            // nodes form a network chain: n2 < n0 < n1 < n2
+            if (test_conf.topology == NetworkTopology.MinimallyConnected)
+                network = [prev_node.address];
+            else
+                network = other_nodes.array;
+        }
 
         Config conf =
         {
             banman : ban_conf,
             node : self,
-            network : test_conf.configure_network ? [prev_node.address] : null,
+            network : assumeUnique(network),
         };
 
         return conf;
     }
 
+    auto keys = WK.Keys.byRange();
+    auto nextKey () { scope (exit) keys.popFront(); return keys.front; }
     NodeConfig[] node_configs;
     Config[] configs;
```

Comment:
I know the name `configs` has been used from start but it looks confusing. The `node_configs` and `configs` were confusing to me as a newcomer. How about using `main_configs` instead of `configs`.

Reply:
I actually had this change in mind too! I'll fix it.